tests/test_graphql_client.py

```diff
@@ -145,6 +145,120 @@ def test_execute_query_with_operation_name(self, post_mock):
         )
 
 
+class TestGraphqlClientExecuteSession(TestCase):
+    """Test cases for synchronous graphql request function using a Session object."""
+
+    def test_execute_basic_query(self):
+        """Sends a graphql POST request to an endpoint."""
+        session_mock = MagicMock()
+        client = GraphqlClient(endpoint="http://www.test-api.com/")
+        query = """
+        {
+            tests {
+                status
+            }
+        }
+        """
+        client.execute(query, session=session_mock)
+
+        session_mock.post.assert_called_once_with(
+            "http://www.test-api.com/", json={"query": query}, headers={}
+        )
+
+    def test_execute_query_with_variables(self):
+        """Sends a graphql POST request with variables."""
+        session_mock = MagicMock()
+        client = GraphqlClient(endpoint="http://www.test-api.com/")
+        query = ""
+        variables = {"id": 123}
+        client.execute(query, variables, session=session_mock)
+
+        session_mock.post.assert_called_once_with(
+            "http://www.test-api.com/",
+            json={"query": query, "variables": variables},
+            headers={},
+        )
+
+    def test_raises_http_errors_as_exceptions(self):
+        """Raises an exception if an http error code is returned in the response."""
+        session_mock = MagicMock()
+        response_mock = MagicMock()
+        response_mock.raise_for_status.side_effect = HTTPError()
+        session_mock.post.return_value = response_mock
+
+        client = GraphqlClient(endpoint="http://www.test-api.com/")
+
+        with self.assertRaises(HTTPError):
+            client.execute(query="", session=session_mock)
+
+    def test_execute_query_with_headers(self):
+        """Sends a graphql POST request with headers."""
+        session_mock = MagicMock()
+        client = GraphqlClient(
+            endpoint="http://www.test-api.com/",
+            headers={"Content-Type": "application/json", "Existing": "123"},
+        )
+        query = ""
+        client.execute(query=query,
+                       headers={"Existing": "456", "New": "foo"},
+                       session=session_mock)
```

Comment:
The linter complains about this line.

```suggestion
        client.execute(
            query=query, headers={"Existing": "456", "New": "foo"}, session=session_mock
        )
```